### PR TITLE
UI modification of the GuessPreviewActivity

### PR DESCRIPTION
**Pull request by leonardvaney**
The changes will include:
- [X] New "..." button with a popup menu instead of the report button.
- [x]  Swipe right and click action to replace the skip and guess buttons.
- [x]  A new button to save the displayed picture on the local gallery of the user.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/modules/app/Wnder.wnder.app.iml" filepath="$PROJECT_DIR$/.idea/modules/app/Wnder.wnder.app.iml" />
```
Comment (JeremyDezalos) [inline .idea/modules.xml:5]: This modification seems undesirable. Why did you do that ?

**Review by JeremyDezalos — CHANGES_REQUESTED** [review]
I have not much to say on top of Romain. Please explain why you changed the .idea file. This caused my gradle build to fail. Also is it just me or can I not swipe left to skip on the emulator ?

File: app/src/main/java/com/github/wnder/OnSwipeTouchListener.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    public boolean onSwipeLeft() {
+        return false;
+    }
+
+    public boolean onSwipeTop() {
+        return false;
+    }
+
+    public boolean onSwipeBottom() {
+        return false;
```
[the remark this answers is not in the document]
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/OnSwipeTouchListener.java:72]: Resolving (explained orally)

**Comment by leonardvaney** [comment]
> 
> 
> I have not much to say on top of Romain. Please explain why you changed the .idea file. This caused my gradle build to fail. Also is it just me or can I not swipe left to skip on the emulator ?

Sorry it's my bad for the swipe. I wrote left but I implement it for a right swipe, I will change the upper description.

File: .idea/modules.xml (hunk 1 above)
Comment (leonardvaney) [reply .idea/modules.xml:5]: I don't know either, I will change it next commit.

File: app/src/main/java/com/github/wnder/OnSwipeTouchListener.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ * The code is entierly based on this discussion from stackoverflow: https://stackoverflow.com/questions/4139288/android-how-to-handle-right-to-left-swipe-gestures
+ */
+public class OnSwipeTouchListener implements View.OnTouchListener {
+    private final GestureDetector gestureDetector;
+
+    public OnSwipeTouchListener(Context ctx){
+        gestureDetector = new GestureDetector(ctx, new GestureListener());
+    }
+
+    public boolean onTouch(final View v, final MotionEvent event) {
```
[the remark this answers is not in the document]
Comment (leonardvaney) [reply app/src/main/java/com/github/wnder/OnSwipeTouchListener.java:21]: > Lint wanrning: Custom view ImageView has setOnTouchListener called on it but does not override performClick

These two warnings concern potential accessibility problems due to the custom touch event, I think they are not important but don't hesitate to tell me back.

**Review by JeremyDezalos — APPROVED** [review]

**Review by rkuenzi-epfl — APPROVED** [review]
Nice improvement since last review!

**Pull request merged**